service/reportsStorage/storage_test.go

```diff
@@ -0,0 +1,659 @@
+		Version:     "V1",
+		Size:        8,
+		Operation:   "w",
+	}
+	rfd1v2 = &models.ReportFileDescriptor{
+		Checksum:    "CHS",
+		ContentType: "text/csv",
+		Created:     time2,
+		Name:        "MjAxOC0wMS0wOVQxMzoxMDowNy4xMjNaLzIwMTgtMDEtMjZUMTU6MTY6MTUuMTIzWg==",
+		Path:        "REPORT_TYPE/MjAxOC0wMS0wOVQxMzoxMDowNy4xMjNaLzIwMTgtMDEtMjZUMTU6MTY6MTUuMTIzWg==/V2",
+		Version:     "V2",
+		Size:        8,
+		Operation:   "w",
+		DataSince:   time1,
+		DataUntil:   time2,
+	}
+	fd1v2 = &storageModels.FileDescriptor{
+		Checksum:    "CHS",
+		ContentType: "text/csv",
+		Created:     time2,
+		Name:        "MjAxOC0wMS0wOVQxMzoxMDowNy4xMjNaLzIwMTgtMDEtMjZUMTU6MTY6MTUuMTIzWg==",
+		Path:        "REPORT_TYPE/MjAxOC0wMS0wOVQxMzoxMDowNy4xMjNaLzIwMTgtMDEtMjZUMTU6MTY6MTUuMTIzWg==/V2",
+		Version:     "V2",
+		Size:        8,
+		Operation:   "w",
+	}
+	rfd2v1 = &models.ReportFileDescriptor{
+		Checksum:    "CHS",
+		ContentType: "text/csv",
+		Created:     time1,
+		Name:        "MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa",
+		Path:        "REPORT_TYPE/MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa/V1",
+		Version:     "V1",
+		Size:        15698,
+		Operation:   "w",
+		DataUntil:   time3,
+	}
+	fd2v1 = &storageModels.FileDescriptor{
+		Checksum:    "CHS",
+		ContentType: "text/csv",
+		Created:     time1,
+		Name:        "MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa",
+		Path:        "REPORT_TYPE/MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa/V1",
+		Version:     "V1",
+		Size:        15698,
+		Operation:   "w",
+	}
+	rfd2v2 = &models.ReportFileDescriptor{
+		Checksum:    "CHS",
+		ContentType: "text/csv",
+		Created:     time3,
+		Name:        "MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa",
+		Path:        "REPORT_TYPE/MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa/V2",
+		Version:     "UUID",
+		Size:        0,
+		Operation:   "d",
+		DataUntil:   time3,
+	}
+	fd2v2 = &storageModels.FileDescriptor{
+		Checksum:    "CHS",
+		ContentType: "text/csv",
+		Created:     time3,
+		Name:        "MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa",
+		Path:        "REPORT_TYPE/MjAxOC0wMS0yOVQxMTowNjo1MS4yMjNa/V2",
+		Version:     "UUID",
+		Size:        0,
+		Operation:   "d",
+	}
+	noErrors   = false
+	withErrors = true
+)
+
+func TestChecksum(t *testing.T) {
+	expected := "7XACtDnprIRfIjV9giusFERzD722AW0-yUMil7nsn3M="
+	svc := service{s3: nil, logger: zerolog.New(os.Stdout)}
+	out, err := svc.Checksum(bytes.NewBuffer([]byte("content")))
+	if out != expected {
+		t.Errorf("Expected %s to equal %s", out, expected)
+	}
+	if err != nil {
+		t.Errorf("Expected err to be nil; got %v", err)
+	}
+}
+
+func TestReportList(t *testing.T) {
+	testCases := []struct {
+		description   string
+		calls         func(*mock.MockStorage) []*gomock.Call
+		expected      []*models.ReportFileDescriptor
+		errorExpected bool
+		exactError    error
+	}{
+		{
+			"BucketExists fails",
+			func(s *mock.MockStorage) []*gomock.Call {
+				return []*gomock.Call{
+					s.EXPECT().BucketExists(gomock.Any(), "REPORT_TYPE").Return(false, fmt.Errorf("Error")),
```

Comment:
invalid operation: s (variable of type *invalid type) has no field or method EXPECT